src/main/java/tc/oc/pgm/community/ReportCommands.java

```diff
@@ -125,35 +144,125 @@ public static void report(
+
+    int perPage = 6;
+    int pages = (reportList.size() + perPage - 1) / perPage;
+
+    Component pageNum =
+        new PersonalizedTranslatable(
+                "command.paginatedResult.page",
+                new PersonalizedText(Integer.toString(page)).color(ChatColor.RED),
+                new PersonalizedText(Integer.toString(pages)).color(ChatColor.RED))
+            .add(ChatColor.AQUA);
+
+    Component header =
+        new PersonalizedTranslatable("moderation.reports.header", headerResultCount, pageNum)
             .getPersonalizedText()
             .color(ChatColor.GRAY);
 
-    // Send message
-    sender.sendMessage(staff);
+    Component formattedHeader =
+        new PersonalizedText(
+            ComponentUtils.horizontalLineHeading(
+                ComponentRenderers.toLegacyText(header, sender), ChatColor.RED));
+
+    new PrettyPaginatedComponentResults<Report>(formattedHeader, perPage) {
+      @Override
+      public Component format(Report data, int index) {
+        Component timeAgo =
+            PeriodFormats.relativePastApproximate(
+                    org.joda.time.Instant.ofEpochMilli(data.getTimeSent().toEpochMilli()))
+                .color(ChatColor.DARK_AQUA);
+        Component hover =
+            new PersonalizedTranslatable("moderation.reports.hover", data.getSenderName())
+                .getPersonalizedText()
+                .color(ChatColor.GRAY);
+        Component formatted =
+            new PersonalizedTranslatable(
+                "moderation.reports.format",
+                timeAgo,
+                data.getTargetName(),
+                new PersonalizedText(data.getReason()).italic(true).color(ChatColor.WHITE));
+
+        return formatted.hoverEvent(Action.SHOW_TEXT, hover.render(sender));
+      }
+    }.display(audience, reportList, page);
+  }
+
+  public static class Report implements Comparable<Report> {
+    private final String id;
+    private final String reason;
+    private final Component targetName;
+    private final Component sender;
+    private final Instant timeSent;
+
+    public Report(String id, String reason, Component targetName, Component sender) {
+      this.id = id;
+      this.reason = reason;
+      this.targetName = targetName;
+      this.sender = sender;
+      this.timeSent = Instant.now();
+    }
+
+    public String getId() {
+      return id;
+    }
+
+    public String getReason() {
+      return reason;
+    }
+
+    public Component getTargetName() {
+      return targetName;
+    }
+
+    public Component getSenderName() {
+      return sender;
+    }
+
+    public Instant getTimeSent() {
+      return timeSent;
+    }
+
+    @Override
+    public int compareTo(Report o) {
+      if (getTimeSent().equals(o.getTimeSent())) {
+        return 0;
+      } else if (getTimeSent().isBefore(o.getTimeSent())) {
+        return 1;
+      } else {
+        return -1;
+      }
```

Comment:
I think this can be simplified to `Instant.compare(getTimeSent(), o.getTimeSent())`

Reply:
Can be simplified to `return timeSent.compareTo(o.timeSent);`